Add support for WebGL2 GetIndexedParameter

mmatyas opened the pull request:
Adds support for the `GetIndexedParameter` WebGL2 call.

See: https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.2

<!-- Please describe your changes on the following line: -->

Depends on #25915.

cc @jdm @zakorgy

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

mmatyas committed:
Add support for WebGL2 GetIndexedParameter

Adds support for the `GetIndexedParameter` WebGL2 call.

See: https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.2

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/script/dom/webidls/WebGL2RenderingContext.webidl, components/script/dom/webgl2renderingcontext.rs, components/script/dom/webglprogram.rs
 * @kichjang: components/script/dom/webidls/WebGL2RenderingContext.webidl, components/script/dom/webgl2renderingcontext.rs, components/script/dom/webglprogram.rs

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

mmatyas commented:
Rebased and fixed the tidy.

jdm commented on tests/wpt/webgl/meta/conformance2/misc/object-deletion-behaviour-2.html.ini:
```diff
@@ -1,5 +1,17 @@
 [object-deletion-behaviour-2.html]
   expected: ERROR
+  [WebGL test #12: gl.isBuffer(bufferBaseTransformFeedback) should be false. Was true.]
```
Judging by https://github.com/servo/servo/blob/52fd6f8aa848699f8f1d40264dd66a765b30dbfa/components/script/dom/webglbuffer.rs#L127, this new failure appears to be caused by the attachment count having a different value after these changes.

mmatyas commented:
You're right, I've forgot to decrement the counters in `DeleteBuffer`. Fixed and updated the tests, the rest of the fails are now related to the transform feedback implementation.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 215dd0a has been approved by `jdm`

<!-- @bors-servo r=jdm 215dd0a1d8a1790c503424a4789fe7a664f896eb -->

bors-servo commented:
:hourglass: Testing commit 215dd0a1d8a1790c503424a4789fe7a664f896eb with merge 12a188ab6110dcff38abc292be0d72a8e437d403...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/JnGIFwmaQiec3lZ-2IymDg)

jdm commented:
@bors-servo retry
* #25513

bors-servo commented:
:hourglass: Testing commit 215dd0a1d8a1790c503424a4789fe7a664f896eb with merge 30fafdca19fd1eeb624c661b4c90c68482a43c05...

bors-servo commented:
:sunny: Test successful - [status-taskcluster](https://community-tc.services.mozilla.com/tasks/groups/DbX8rD1GTLOhr_k_mG-Dgg)
Approved by: jdm
Pushing 30fafdca19fd1eeb624c661b4c90c68482a43c05 to master...
